server/src/test/java/org/elasticsearch/action/admin/indices/rollover/RolloverRequestTests.java

```diff
@@ -141,4 +154,88 @@ public void testSerialize() throws Exception {
             }
         }
     }
+
+    public void testToAndFromXContent() throws IOException {
+        RolloverRequest rolloverRequest = createTestItem();
+
+        final XContentType xContentType = randomFrom(XContentType.values());
+        boolean humanReadable = randomBoolean();
+        BytesReference originalBytes = toShuffledXContent(rolloverRequest, xContentType, EMPTY_PARAMS, humanReadable);
+
+        RolloverRequest parsedRolloverRequest = new RolloverRequest();
+        parsedRolloverRequest.fromXContent(createParser(xContentType.xContent(), originalBytes));
+
+        CreateIndexRequest createIndexRequest = rolloverRequest.getCreateIndexRequest();
+        CreateIndexRequest parsedCreateIndexRequest = parsedRolloverRequest.getCreateIndexRequest();
+        CreateIndexRequestTests.assertMappingsEqual(createIndexRequest.mappings(), parsedCreateIndexRequest.mappings());
+        CreateIndexRequestTests.assertAliasesEqual(createIndexRequest.aliases(), parsedCreateIndexRequest.aliases());
+        assertEquals(createIndexRequest.settings(), parsedCreateIndexRequest.settings());
+
+        assertEquals(rolloverRequest.getConditions(), parsedRolloverRequest.getConditions());
+
+        BytesReference finalBytes = toShuffledXContent(parsedRolloverRequest, xContentType, EMPTY_PARAMS, humanReadable);
+        ElasticsearchAssertions.assertToXContentEquivalent(originalBytes, finalBytes, xContentType);
+    }
+
+    public void testUnknownFields() throws IOException {
+        final RolloverRequest request = new RolloverRequest();
+        XContentType xContentType = randomFrom(XContentType.values());
+        final XContentBuilder builder = XContentFactory.contentBuilder(xContentType);
+        builder.startObject();
+        {
+            builder.startObject("conditions");
+            builder.field("max_age", "10d");
+            builder.endObject();
+        }
+        builder.endObject();
+        BytesReference mutated = XContentTestUtils.insertRandomFields(xContentType, builder.bytes(), null, random());
+        expectThrows(ParsingException.class, () -> request.fromXContent(createParser(xContentType.xContent(), mutated)));
+    }
+
+    public void testAddConditions() {
```

Comment:
Maybe add a comment that this test is about failing when calling addXYZ twice, or rename the test method to reflect that better

Reply:
I am renaming to `testSameConditionCanOnlyBeAddedOnce`